### ISSUE TITLE
Implement models done in homeworks

### ISSUE THREAD

**Issue by malangadan** [malangadan/Challengers#6]
Logistic regression, SVM gaussian kernel, SVM linear kernel, Decision tree, Random Forest, KNN, ball Tree, and kd tree should be implemented. This should not be difficult because this has been done in the homework already and you can just use your code from there.

The difficult in this issue is tuning the hyperparamaters. If you are doing this issue please demonstrate a difference between before and after hypertuning. Some models will perform better on the data but that is expected. 

Make sure to to check all metrics with testing data and training data. This means outputting f1 score, precision, and accuracy for every model based on the training data and testing data.
